### DISCUSSION TITLE
Adding backface-visibility?

### DISCUSSION THREAD

**Discussion by Ratko-Solaja** [tailwindlabs/tailwindcss#1686] [General]
What do you think about adding ```backface-visibility``` property that would function just like the ```visibility```:

```
.backface-visible {
  backface-visibility: visible;
}

.backface-invisible {
  backface-visibility: hidden;
}
```

**Comment by lucashfreitas** [comment]
up

**Comment by thomasnavarro** [comment]
Hi! 
Add this to tailwind.config.js and use `<div class="backface-hidden"></div>`

```
const plugin = require('tailwindcss/plugin')

const backfaceVisibility = plugin(function({addUtilities}) {
  addUtilities({
    '.backface-visible': {
      'backface-visibility': 'visible',
    },
    '.backface-hidden': {
      'backface-visibility': 'hidden',
    }
  })
});

module.exports = {
  plugins: [backfaceVisibility]
}
```

**Reply by mohsin** [reply]
Ideally, you'd want to target older browser versions like this:
```
const backfaceVisibility = plugin(function({addUtilities}) {
  addUtilities({
    '.backface-visible': {
      'backface-visibility': 'visible',
      '-moz-backface-visibility': 'visible',
      '-webkit-backface-visibility': 'visible',
      '-ms-backface-visibility': 'visible'
    },
    '.backface-hidden': {
      'backface-visibility': 'hidden',
      '-moz-backface-visibility': 'hidden',
      '-webkit-backface-visibility': 'hidden',
      '-ms-backface-visibility': 'hidden'
    }
  })
});
```

**Reply by mybearworld** [reply]
Shouldn't autoprefixer do this for you?

**Comment by max-sym** [comment]
Is this going to be implemented? Pretty relevant, I use it a lot in places that have `opacity` + `transform` animations to make it smooth.

**Reply by power-f-GOD** [reply]
Same here.

**Comment by CRIMSON-CORP** [comment]
I hope this get implemented soon, it really is a good utility to add

**Comment by podrivo** [comment]
I was trying to use it for a css flip animation. I hope this gets implemented! 🤞 
@thomasnavarro @mohsin Thank you for the solution! 🙌

**Comment by Milkywayrules** [comment]
anyone knows the update? or it is won't be implemented? didn't find any official statement from the Tailwind Team itself and this discussion is pretty dead? *_*

**Reply by podrivo** [reply]
@Milkywayrules I would just use an alternative solution presented above.
You can also join the conversation on this PR: #10982

**Comment by GG-TOF** [comment]
Up
